src/pages/homeCollection/kitAssembly.js

```diff
@@ -152,15 +152,20 @@ const processAssembledKit = () => {
     saveKitButton.addEventListener('click', async () => { 
       let kitObj = {};
 
-      const scannedBarcodeValue = document.getElementById('scannedBarcode').value.trim();
-      const supplyKitIdValue = document.getElementById('supplyKitId').value.trim();
-      const returnKitIdValue = document.getElementById('returnKitId').value.trim();
-      const collectionCupIdValue = document.getElementById('cupId').value;
-      const collectionCardIdValue = document.getElementById('cardId').value;
-
+      const scannedBarcodeValue = (document.getElementById('scannedBarcode').value.trim() !== undefined) 
+      ? document.getElementById('scannedBarcode').value.trim() : 0;
+      const supplyKitIdValue = (document.getElementById('supplyKitId').value.trim() !== undefined) 
+      ? document.getElementById('supplyKitId').value.trim() : 0;
+      const returnKitIdValue = (document.getElementById('returnKitId').value.trim() !== undefined) 
+      ? document.getElementById('returnKitId').value.trim() : 0;
+      const collectionCupIdValue = (document.getElementById('cupId').value.trim() !== undefined) 
+      ? document.getElementById('cupId').value.trim() : 0;
+      const collectionCardIdValue = (document.getElementById('cardId').value.trim() !== undefined) 
+      ? document.getElementById('cardId').value.trim() : 0;
```

Comment:
Suggestions:
- Query each element and get value only once instead of 2 times
- Use optional chaining (eg `document.getElementById('scannedBarcode')?.value?.trim()`) to avoid running errors: elements may not be found